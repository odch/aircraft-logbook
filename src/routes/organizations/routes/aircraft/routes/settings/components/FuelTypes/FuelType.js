import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl } from 'react-intl'
import ListItemText from '@material-ui/core/ListItemText'
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
import IconButton from '@material-ui/core/IconButton'
import ListItem from '@material-ui/core/ListItem'
import Menu from '@material-ui/core/Menu'
import MenuItem from '@material-ui/core/MenuItem'
import MoreVertIcon from '@material-ui/icons/MoreVert'
import { fuelType as fuelTypeShape } from '../../../../../../../../shapes/aircraft'
import { intl as intlShape } from '../../../../../../../../shapes'

class FuelType extends React.Component {
  state = {
    anchorEl: null
  }

  handleMenuOpen = event => {
    this.setState({ anchorEl: event.currentTarget })
  }

  handleMenuClose = () => {
    this.setState({ anchorEl: null })
  }

  handleMenuItemClick = (callback, args) => () => {
    if (typeof callback === 'function') {
      callback(...args)
    }
    this.handleMenuClose()
  }

  msg = id => this.props.intl.formatMessage({ id })

  render() {
    const { fuelType } = this.props
    return (
      <React.Fragment>
        <ListItem disableGutters>
          <ListItemText
            primary={fuelType.description}
            secondary={fuelType.name}
          />
          <ListItemSecondaryAction>
            <IconButton onClick={this.handleMenuOpen}>
              <MoreVertIcon />
            </IconButton>
          </ListItemSecondaryAction>
        </ListItem>
        {this.state.anchorEl && this.renderMenu()}
      </React.Fragment>
    )
  }

  renderMenu = () => {
    const { fuelType, openDeleteFuelTypeDialog } = this.props

    return (
      <Menu anchorEl={this.state.anchorEl} onClose={this.handleMenuClose} open>
        <MenuItem
          onClick={this.handleMenuItemClick(openDeleteFuelTypeDialog, [
            fuelType
          ])}
        >
          {this.msg('aircraft.settings.fueltype.delete')}
        </MenuItem>
      </Menu>
    )
  }
}

FuelType.propTypes = {
  fuelType: fuelTypeShape.isRequired,
  intl: intlShape.isRequired,
  openDeleteFuelTypeDialog: PropTypes.func.isRequired
}

export default injectIntl(FuelType)
