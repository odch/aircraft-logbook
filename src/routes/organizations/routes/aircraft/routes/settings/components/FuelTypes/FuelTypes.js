import React from 'react'
import { FormattedMessage } from 'react-intl'
import Typography from '@material-ui/core/Typography'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import { fuelTypes as fuelTypesShape } from '../../../../../../../../shapes/aircraft'

class FuelTypes extends React.Component {
  render() {
    const { types } = this.props

    return (
      <div>
        <Typography variant="title" gutterBottom>
          <FormattedMessage id="aircraft.settings.fueltypes" />
        </Typography>
        {types.length > 0 ? (
          <List dense>
            {types.map(type => (
              <ListItem key={type.name} disableGutters>
                <ListItemText
                  primary={type.description}
                  secondary={type.name}
                />
              </ListItem>
            ))}
          </List>
        ) : (
          <Typography paragraph>
            <FormattedMessage id="aircraft.settings.fueltypes.none" />
          </Typography>
        )}
      </div>
    )
  }
}

FuelTypes.propTypes = {
  types: fuelTypesShape.isRequired
}

export default FuelTypes
