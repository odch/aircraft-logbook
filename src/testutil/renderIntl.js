import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
