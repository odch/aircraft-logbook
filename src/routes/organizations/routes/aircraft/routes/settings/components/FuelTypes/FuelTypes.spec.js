import React from 'react'
import renderIntl from '../../../../../../../../testutil/renderIntl'
import FuelTypes from './FuelTypes'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('routes', () => {
          describe('settings', () => {
            describe('components', () => {
              describe('FuelTypes', () => {
                it('renders fuel types', () => {
                  const tree = renderIntl(
                    <FuelTypes
                      types={[
                        { name: 'avgas', description: 'AvGas' },
                        { name: 'mogas', description: 'MoGas' }
                      ]}
                      deleteFuelTypeDialog={{
                        open: false
                      }}
                    />
                  ).toJSON()
                  expect(tree).toMatchSnapshot()
                })

                it('renders no types message', () => {
                  const tree = renderIntl(
                    <FuelTypes
                      types={[]}
                      deleteFuelTypeDialog={{
                        open: false
                      }}
                    />
                  ).toJSON()
                  expect(tree).toMatchSnapshot()
                })
              })
            })
          })
        })
      })
    })
  })
})
