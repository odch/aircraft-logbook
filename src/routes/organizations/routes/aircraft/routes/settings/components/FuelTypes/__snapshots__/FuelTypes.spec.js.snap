// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft routes settings components FuelTypes renders fuel types 1`] = `
<div>
  <h5
    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
  >
    Treibstofftypen
  </h5>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Treibstofftyp hinzufügen
    </span>
  </button>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding"
  >
    <li
      className="MuiListItem-root MuiListItem-dense"
      disabled={false}
    >
      <div
        className="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          AvGas
        </span>
        <p
          className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          avgas
        </p>
      </div>
    </li>
    <li
      className="MuiListItem-root MuiListItem-dense"
      disabled={false}
    >
      <div
        className="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          MoGas
        </span>
        <p
          className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          mogas
        </p>
      </div>
    </li>
  </ul>
</div>
`;

exports[`routes organizations routes aircraft routes settings components FuelTypes renders no types message 1`] = `
<div>
  <h5
    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
  >
    Treibstofftypen
  </h5>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Treibstofftyp hinzufügen
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
  >
    Keine Treibstofftypen erfasst
  </p>
</div>
`;
